### ISSUE TITLE
LANGUAGE.CSV overhaul

### ISSUE THREAD

**Issue by MuadDib90** [Realm667/WolfenDoom#431]
First of all, please do not close this thread, until I finish the overhaul of language.csv. I will try to go through it and report here inconsistencies (from my point of view), if I find any, to be checked (especialy by @Tormentor667, but also others) and to decide, what to do next. I will be gradually adding here sections.

- [x] **Outset -- [CLOSED]**
<details>
  <summary>From the very begging until "mission objectives" section:</summary>

.
I still need to clarify some things for CSY translation.
_[CHECKED]_ What does this mean/what is this for:
- "stop recovery"
- "stop treshold"
- "droplets"

Now for the "things":

- _[CHECKED]_ **QUITMSG1** 
\cCIn case you hadn't noticed, there is a Nazi invasion going on. (Shouldn't the first sentence be in present perfect instead of past perfect?)
- _[CHECKED]_ **TXT_YES; TEXT_NO** 
This is not shown (their boxes or something), when quitting the game. Maybe it IS there, but overlaid by something, so it is not  visible?
- _[CHECKED]_ **AM_ITEMS**
Should be "treasures", plural.
- _[CHECKED]_ I seem not to be able to find many of the settings (identifiers) listed in "mainmenu items" section, when entering game settings. Is it really all there?

</details>

- [x] **Mission objectives -- [CLOSED]**
<details>
  <summary>Section:</summary>

.
- _[CHECKED]_ **MO_ADDED** "new objective added"
Could be displayed 2 times longer, than is is now. Just a cosmetic for more convenience.
- _[CHECKED]_ The text of objectives to enter the secret missions as "collect files eisenmann" and "find mayan artifacts" have no unified appearence. Will be good if they could be always the first or the last of "secondary objectives".
- _[CHECKED]_ **MO_C3M2_P3** "destroy 3 objectives"
I think there could be better "Take care of 3 objectives"
- _[CHECKED]_ Can I ask for Czech language, if you could add back the smaller frame for objectives? For the same reason it was left there for English. I was able to fit all objectives into it and it looks much better.

</details>

- [x] **Various -- [CLOSED]**
<details>
  <summary>After "mission objectives" until "briefing messages" section:</summary>

.
- _[CHECKED]_ **KAR98K, PYRO, STEN, TESLAC, UMG43**
Weapon pickups messages of these are too short. They contain additional information regarding alt-fire usage. Should be longer displayed, so players have time to read them.

> It can likely be done by changing the default value of the ```con_notifytime``` CVAR (it is 3 seconds now), but it will affect every message, not just these ones. Maybe placing the altfire hints into the pickup messages was not the best idea of mine, but I don't see how to do it better. Also, the player can open the console and read them from there. --N00b, 28 Mar 2021
>> If it can be read in the console afterwards, then it can be considered as solved, I guess. But do players know about this feature? If so... checked for me. --Muad, 28 Mar 2021
> Well, some do, and some very likely do not... But I would compare this engine feature with jumping/crouching in the following way. An 'old generation' Doom player who stumbled upon GZDoom for the first time (and who knows that it's a Doom port) probably would not have expected that you can jump or duck. Newer FPS games pretty much all use consoles, so it should not be too much of a problem for younger players. --N00b, 28 Mar 2021

</details>

- [x] **Briefing messages -- [CLOSED]**
<details>
  <summary>Briefing messages section:</summary>

.
- _[CHECKED]_ **BRIEFINGC2M5_2**
There is a talk "we've destroyed the Eisenmann facility". But how can be known, if the player did the C1M0 mission? I think it can be not. You can also go play the 2nd charpter right away. So... are we simply "counting" with it being destroyed in this "story position"? (not thinking about if the player did the C1M0 mission, or consequences between charpters ect...)

> Recently (maybe some months ago) there was a discussion about that, and IIRC it was established that destroying C1M0 is 'canonical'. So, exactly what you say here. --N00b, 28 Mar 2021
>> Yes, I remember that. I wrote it mainly as "to be sure" statement. - Then this is checked. --Muad, 28 Mar 2021

- _[CHECKED]_ **BRIEFINGC2M0_6 & BRIEFINGC2M0_7**
Gen. Miller tells you on briefing that Douglas can't go with you on this mission, as he has duties regarding the traitor. Yet he is standing next to the truck and talks to you, as he is to "send you in". (If you think about it, he wouldn't be there, as he would have "full hands with the traitor".) Would be better if (for this one instance) there was standing Ascher, telling you about Douglas can't come, and he is sorry and is a substitute for this one mission.

> Maybe the Douglas actor should be replaced with Ascher when you activate the secret mission from Gutenberg. --N00b, 28 Mar 2021
>> Yes, that is good, also we need/have to replace those 2 the messages. Former Douglas, now from Ascher. --Muad, 28 Mar 2021

- _[CHECKED]_ **BRIEFINGC3M1_4 & BRIEFINGC3M1_7**
Here they (Miller and Ascher) are talking about "steal a radio" or "get a radio", yet they can't know the actual situation on the battlefield. (It it written that way, because someone, who wrote those lines, "knew how the mission looks and works", what are those objectives.)
.
I'd recommend to change what Miller says FROM ~If you disable the Nazis' communications and steal a radio, Stg Ascher will help you to get past the guards and rescue the prisoners.~ TO _When inside, find a way to get in touch with Ascher, then he will help you to get past the guards and rescue the prisoners._
.
Also Ascher's line to be changed FROM ~...at 0800, and I'll update you as soon as you get a radio from the camp.~ TO _...at 0800. Let's pray you'll find a radio there, then I'll update you._ (Or something similar to this one... you get the idea.)
.
CAUTION, other mission texts (dialogs with other characters) MAY need to be changed accordingly to this change, donno now!

> This is also fair point, good suggestions! --N00b, 28 Mar 2021
</details>

- [x] **Epilogues (strife texts) -- [CLOSED]**

Nothing found.

- [x] **Douglas messages -- [CLOSED]**

Only cosmetic repairs. Spacing ect.

- [x] **Ryan, Big Dave and Ascher messages -- [CLOSED]**
<details>
  <summary>Section:</summary>

.
- _[CHECKED]_ **PRISONERMESSAGE07**
There is sentence: _you managed to escape all the trouble I prepared, and survive all the traps I set_ -- shouldn't here be the past simpe "SURVIVED"?

> Or '...prepared, and to survive all...'? --N00b, 28 Mar 2021
>> Sure, or that one. --Muad, 28 Mar 2021

- _[CHECKED]_ **"willy boy" phrase**
I suggest to use this phrase in those messages from Ryan only after he kills Douglas. This phrase evokes (at least in me) the sarcasm. But I can't really tell how does that sound to en english native speaker...
Concerned identifiers where it should be removed, if it happens to be the case:
   - PRISONERMESSAGE13
   - PRISONERMESSAGE16
   - PRISONERMESSAGE19
   - PRISONERAGENTCONV16
   - PRISONERAGENTCONV21

> Why not have these sarcastic parts? I think the style adds some distinctive personality to Ryan. Do you think it is inappropriate for some reason? --N00b, 28 Mar 2021
>> I wouldn't use "inappropriate" exactly here. Sure distinctiveness is good. If it is meant to be as kind of Ryan's "catchphrase", then I guess it is OK. But I got a feeling it was extremely sarcastic from _evil Ryan_'s mouth. And maybe he wants to hide this way the truth about him... Maybe that's why it feels odd to me to hear this from _normal Ryan_. --Muad, 28 Mar 2021
> Ryan is only 'externally' normal, his failure, imprisonment and inducement occured before the events of Chapter I. He has been a traitor since the beginning of Blade of Agony. --N00b, 28 Mar 2021

- _[CHECKED]_ **GUTENBERGCONVO04 & GUTENBERGCONVO06**
Are those really unused? If so, please place UNSUSED LINE placeholder in ENG column.

- _[CHECKED]_ **GUTENBERGCONVO10**
Here is: _\cCWhat's a German scientist doing in an \cCAllied lab?_ -- Should the Allied lab be some specific color, or can "\cC" be removed for redundancy?

> I'd remove the color code. --N00b, 28 Mar 2021
>> Agreed. --Muad, 28 Mar 2021

- _[CHECKED]_ **MARINE2CONV16**
In the sentence: _He told me about his wife, his children, his plans after we have won the war._ -- Shouldn't the last part after the word "after" be in some kind of tense condition?

> Not an expert in English grammar, but here is some interesting [evidence](https://books.google.com/ngrams/graph?content=after+we+have+won&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cafter%20we%20have%20won%3B%2Cc0) from Google Ngrams which shows that this form is correct. --N00b, 28 Mar 2021
>> Wow, then it's solved. --Muad, 28 Mar 2021

- _[CHECKED]_ **MARINEAFTERDOUGLESCONV03**
I can't really remember, where this is displayed, so just for peace of mind, should there be at the end of the paragraph "...proceed with \cQCpt. Blake\cC!" ? (Is it really Blake or someone else?)

- _[CHECKED]_ **ASCHERCONV20**
Here is: _The B-24 support has been our joker, but the \cRradio tower\cC northwest of here is on higher ground, and it's preventing them from pushing us further._
.
The thing is, either "flak cannons" can prevent B-24 bombers from GETTING THERE (physically) to help them (soldiers) push further, or "radio tower" can prevent THE CONTACT between soldiers on battleground and those B-24 bombers.
Can the difference be seen?
.
And if it will be changed, we need to also directly change ASCHERCONV20. -- "We will be able to contact B-24 bombers, so they are able to destroy enemy's defeses."

> The tower is preventing them indirectly, it coordinates the anti-aircraft fire. --N00b, 28 Mar 2021
>> Yes, I know what you mean. In czech we have also this kind of sentence structure, but it just sounds very odd to me in english. But I trust you in this. --Muad, 28 Mar 2021

- _[CHECKED]_ **ASCHERCONV42**
Here: _This time we will hunt him down, dead or alive, BUT this war, this madness ends here, it ends today._ -- Shouldn't be better to use AND intead of BUT?

</details>

- [x] **Rest of the messages/conversations -- [CLOSED]**
<details>
  <summary>Section:</summary>

.
- _[CHECKED]_ **BARKEEPERCONV12** 
Here could be a pretty reference, and another character development for that instance. :)
I'd change the sentence: _Makes you look keen... \n\n..._ TO _Makes you look keen... like 'a Commander', hihi... oh, come on, zmile for me a bit! -- Oki, never mind...\n\n..._

- _[CHECKED]_ **BARRANDICONV11**
Shouldn't in: _...I would have said \cPLaz\cC put a few of his mushrooms..._ be TO PUT? (or change _his_ to _your_)

> The original line should likely be equivalent to 'I would have said that Laz consumed...', not 'I ordered Laz to...' --N00b, 28 Mar 2021

- _[CHECKED]_ **CLERKCONV12**
I'd add a word "instantly" after "Health +25"

- _[CHECKED]_ **VOLKOFFCONV05**
A the end of the whole paragraph, shouldn't be the _Führerbunker_ also with \cB color mark?

</details>

- [x] **Various - counter texts, hint texts ect. -- [CLOSED]**
<details>
  <summary>Section:</summary>

.
- _[CHECKED]_ **C3M0ABONUSS7 & C3M0ABONUSS8** 
Sorry to reopen this again, but I am not sure if there wasn't a misunderstanding or not, so to be sure...
The ultimate gift for completing the "kill 300 zombies" objective should be the Nebelwerfer, shouldn't it? Thus it should be the last entry in the list, so the C3M0ABONUSS8. And the last that it is now "UMG and 9mm ammo" should be removed. The one reason for this is, you get only ONE super weapon for completing the objective - the nebelwerfer. Second reason, you can get the UMG already later on in zone 4: the city; in that corner guarded by landmines (remember?). So the 7th entry could by some gold for example.

>Will not be changed since the need in script change a few days before the release. --MuadDib, 4 Apr 21

- _[CHECKED]_ **SAFEHINT**
~I'd recommend~ I need to let it be displayed longer time. Especially in Czech I need 2 times longer text to properly explain what to do to prevent players' frustration on this. I dare to say 3 times longer to be displayed (for Czech at least).

> Is it displayed again when the minigame is reopened? --N00b, 28 Mar 2021
>> Wait, what? Minigame? Are we talking about the same? This is the hint that tells to a player how to open the safe with Commande Keen ken karte. How to use that dial. --Muad, 28 Mar 2021
> Yes, exactly that. Also we should likely add more detailed explanations to other languages, not everyone knows e. g. that the sequence of directions you should turn the dial in (left, right, left... or the other way around? :-> ) is fixed. --N00b, 28 Mar 2021
>> Yes, I kinda described it there, that's why it needs to be displayed longer time. There are those arrows that tell the first, second and third turn though. --Muad, 28 Mar 2021

- _[CHECKED]_ **CUTSCENE1 & CUTSCENE2 & CUTSCENE2**
Those entries do not need any color prefix? (they don't have at present - any other have, so...)

> EDIT: sorry, it should have been "_CUTSCEN1_" ...2 and 3.

</details>

- [x] **Text papers. -- [CLOSED]**

Nothing worth mentioning. Just typos here...

- [x] **Secrets, credits. -- [CLOSED]**
<details>
  <summary>Section:</summary>

.
- _[CHECKED]_  **SECRET323**
Can't we better say here _"building"_ with commas? Because it is, in fact, a ruin; but saying "a ruin" would probably be too easy.

- _[CHECKED]_ **CREDITSM01-...11**
I don't remember these messages being displayed anywhere. Are they redundant, or still waiting to be coded for displaying at the end?

</details>

- [x] **Info pages. -- [CLOSED]**

Nothing found.

- [x] **Developer commentary. -- [CLOSED]**
<details>
  <summary>Section:</summary>

.
- _[CHECKED]_ **DEVC3M4_1** @Username-N00b-is-not-available 
I can't undestand the last sentence. Particulary the the part _in secrets and not_. Can you please try to explain it differently?
The sentence: _I took measures to add bits of gold here and there, in secrets and not, and the this value increased threefold._

> Some of the gold that was put into the maps by me during this process is hidden in secrets, and some just lies in plain sight (e. g. on officers'/scientists' tables). --N00b, 04 Apr 2021

- _[CHECKED]_ **DEVC3M5B_3** @Talon1024 
I'd remove the word _recently_. It won't be "recently" anymore, when players will be (finaly) able to play the game. Also, should't there be "players' "?
The sentence: _Early versions of several Blade of Agony maps, and maps that were recently changed according to player feedback..._

- _[CHECKED]_ **DEVC3M5C_1** @Tormentor667 
Although a cosmetic change. I'd change the last sentence FROM _Welcome to the airship._ TO _Welcome aboard!_

</details>

**Comment by Tormentor667** [comment]
Count myself out when it comes to proper english grammar - I can help with context questions though.

**Comment by AFADoomer** [comment]
QUITMSG1 is correct either way.
TXT_YES and TXT_NO are leftovers from the default exit screens, I think, which have been replaced.
AM_ITEMS should stay like it is.  "Treasure" is correct in this context - just like you would say "money" and not "monies".

**Comment by MuadDib90** [comment]
> 
> 
> Count myself out when it comes to proper english grammar - I can help with context questions though. When it comes to settings, best ask @Ozymandias81, he did most of the options ingame.

Yes, exactly, that's what I meant. We need you in helping with the context, because your vision started it all... so, I think, you should have the final word.
About that ingame settings, OK, will do. @Ozymandias81, my friend, can you please help me here?

> 
> 
> QUITMSG1 is correct either way.
> TXT_YES and TXT_NO are leftovers from the default exit screens, I think, which have been replaced.
> AM_ITEMS should stay like it is. "Treasure" is correct in this context - just like you would say "money" and not "monies".


@AFADoomer, thanks, that clarifies this.

**Comment by MuadDib90** [comment]
Answered @Username-N00b-is-not-available's comments.
3 sections remaining...

Any other comments? @Tormentor667 @Talon1024 @Ozymandias81 @AFADoomer

**Comment by Tormentor667** [comment]
> MO_C3M2_P3 "destroy 3 objectives" I think there could be better "Take care of 3 objectives"

I changed this into "Complete 3 objectives" which seems right to me.

> I'd recommend to change what Miller says FROM _If you disable the Nazis' communications and steal a radio, Stg Ascher will help you to get past the guards and rescue the prisoners_. TO _When inside, find a way to get in touch with Ascher, then he will help you to get past the guards and rescue the prisoners_.

I agree here, this change makes sense.

> I wouldn't use "inappropriate" exactly here. Sure distinctiveness is good. If it is meant to be as kind of Ryan's "catchphrase", then I guess it is OK. But I got a feeling it was extremely sarcastic from evil Ryan's mouth. And maybe he wants to hide this way the truth about him... Maybe that's why it feels odd to me to hear this from normal Ryan. --Muad, 28 Mar 2021

I wouldn't consider it "sarcastic" but some kind of arrogant, which fits the character well.

> GUTENBERGCONVO04 & GUTENBERGCONVO06 Are those really unused? If so, please place UNSUSED LINE placeholder in ENG column.

I think they _are_ used after you give all files/artifacts to Gutenberg - at least I wrote them and added them some months and years ago.

> MARINEAFTERDOUGLESCONV03 I can't really remember, where this is displayed, so just for peace of mind, should there be at the end of the paragraph "...proceed with \cQCpt. Blake\cC!" ? (Is it really Blake or someone else?)

This is all fine, it is from the beginning of C2M4 afaik (the operation in Mahdia), checking the buildings before you jump on the truck and move on with Douglas.

> ASCHERCONV42 Here: This time we will hunt him down, dead or alive, BUT this war, this madness ends here, it ends today. -- Shouldn't be better to use AND intead of BUT?

Agreed and fixed.

> BARKEEPERCONV12 Here could be a pretty reference, and another character development for that instance. :)
I'd change the sentence: Makes you look keen... \n\n... TO Makes you look keen... like 'a Commander', hihi... oh, come on, zmile for me a bit! -- Oki, never mind...\n\n...

Smart idea, I love it, so please add it :)

> VOLKOFFCONV05 A the end of the whole paragraph, shouldn't be the Führerbunker also with \cB color mark?

Agreed and fixed.

> CUTSCENE1 & CUTSCENE2 & CUTSCENE2 Those entries do not need any color prefix? (they don't have at present - any other have, so...)

These can be added, I am okay with it.

**Comment by MuadDib90** [comment]
Thanks for comments @Tormentor667.
What I thought I can dare to repair by myself, is done.

And I must say I am damn good! I succeeded in translading the new Juliette's line to czech and somehow keep the point. Yay! :D

**Comment by MuadDib90** [comment]
Now only dev_coms are left to check... let's ge to it!

**Comment by Tormentor667** [comment]
I am so excited, great work on all that @MuadDib90

**Comment by MuadDib90** [comment]
Thank you. You're welcome @Tormentor667 
Just helping a bit. :)

**Comment by Tormentor667** [comment]
> The text of objectives to enter the secret missions as "collect files eisenmann" and "find mayan artifacts" have no unified appearence. Will be good if they could be always the first or the last of "secondary objectives".

That would mean a lot of work as not only the texts need to be relocated but also the mission objectives in each map. At this current stage of development and with the release nearing in less than a month, I'd check this as "low priority".

> **BRIEFINGC2M0_6 & BRIEFINGC2M0_7**
Gen. Miller tells you on briefing that Douglas can't go with you on this mission, as he has duties regarding the traitor. Yet he is standing next to the truck and talks to you, as he is to "send you in". (If you think about it, he wouldn't be there, as he would have "full hands with the traitor".) Would be better if (for this one instance) there was standing Ascher, telling you about Douglas can't come, and he is sorry and is a substitute for this one mission.

Instead of fiddeling around with the scripts in this particular case, I'd suggest changing the texts of Miller and Douglas somehow, that explains either that Douglas can't help you for a different reason or that Douglas will just send you in but then returning for a different duty. But let's avoid changing the INTERMAP scripts at any cost for the same reason as the other issue. 

Same goes to **BRIEFINGC3M1_4 & BRIEFINGC3M1_7** and **CLERKCONV12** - text changes are okay, script changes not.

> **CREDITSM01-...11**
I don't remember these messages being displayed anywhere. Are they redundant, or still waiting to be coded for displaying at the end?

These are displayed in the epilogue after the last map, the ending sequence. You can play it by warping to CREDITSM

**Comment by MuadDib90** [comment]
I understand.
Will try to prepare something for approval...

**Comment by MuadDib90** [comment]
With adding the last missing section of "developer commentary", the work on language.csv overhaul is complete from my side.

Now only resolving the rest remains... and we can close this store. :)

**Comment by MuadDib90** [comment]
- REGARDING **SAFEHINT**

I'd add this:
_\cCRotate alternately as red arrows indicate using \cU< arrow keys >\cC to values of the safe combination.\n\cCThen press \cU< enter >\cC to open the safe._

(Without the spaces in < >)

- REGARDING **BRIEFINGC2M0_6 & BRIEFINGC2M0_7**

@Talon1024 proposed a line with the change in bold:
_Unfortunately, Cpt. Douglas has other duties regarding the traitor to attend to, so **he can only take you as far as the docks, where** we have a captured U-boat waiting to take you to the excavation site. Now, go gear yourself up as best you can at Lt. Rojas' shop, and make haste! Godspeed, Will. Dismissed!_

And I agree.

- REGARDING **BRIEFINGC3M1_4 & BRIEFINGC3M1_7**

Only changes that need to be done are:

Original C3M1_4:
_\cCYou need to find a way inside, destroy the camp buildings, and free as many prisoners as possible before \cRHitler\cC's plans succeed. If you disable the Nazis' communications and steal a radio, \cQSgt. Ascher\cC will help you get past the guards and rescue the prisoners._

New C3M1_4:
\cCYou need to find a way inside, destroy the camp buildings, and free as many prisoners as possible before \cRHitler\cC's plans succeed. When inside, find a way to get in touch with \cQSgt. Ascher\cC, then he will help you get past the guards and rescue the prisoners.

Original C3M1_7:
_\cCI'll drop you off near the train station at 0800, and I'll update you as soon as you get a radio from the camp. Watch out for surprises, \cPCaptain\cC. \cBSchattenau\cC must be more than just an ordinary concentration camp if the rumors are true. Good luck, let's move out!_

New C3M1_7:
\cCI'll drop you off near the train station at 0800. Let’s pray you’ll find a radio there, then I’ll update you. Watch out for surprises, \cPCaptain\cC. \cBSchattenau\cC must be more than just an ordinary concentration camp if the rumors are true. Good luck, let's move out!

In the camp BJ will tell the prisoner, that he need to find a radio. But that’s OK since in C1M1_7 Ascher is speaking of it.

---

**If you all agree with these, and provide feedback on what to do with last section, dev_coms, I will take care of it, and we can close this.**

**Comment by Talon1024** [comment]
> > GUTENBERGCONVO04 & GUTENBERGCONVO06 Are those really unused? If so, please place UNSUSED LINE placeholder in ENG column.
> 
> I think they _are_ used after you give all files/artifacts to Gutenberg - at least I wrote them and added them some months and years ago.

They are unused. The lines which Prof. Gutenberg says after you hand in all of the Eisenmann files or Mayan artifacts are GUTENBERGCONVO19 and GUTENBERGCONVO21 respectively.

**Comment by Tormentor667** [comment]
Agreed with all of them.

**Comment by MuadDib90** [comment]
> 
> 
> Agreed with all of them.

OK, I am on it.

**Comment by MuadDib90** [comment]
Everything is done. Closing this thread.

**Comment by Tormentor667** [comment]
Thanks alot for the effort!

**Comment by MuadDib90** [comment]
You are welcome.
